rust/src/lib/iface.rs

```diff
@@ -440,26 +470,21 @@ impl Interface {
             Self::Ethernet(ref mut iface) => {
                 iface.pre_verify_cleanup();
             }
-            Self::Vlan(ref mut iface) => {
-                iface.pre_verify_cleanup();
-            }
-            Self::Unknown(ref mut iface) => {
-                iface.base.pre_verify_cleanup();
-            }
-            Self::Dummy(ref mut iface) => {
-                iface.base.pre_verify_cleanup();
-            }
-            Self::OvsInterface(ref mut iface) => {
-                iface.base.pre_verify_cleanup();
-            }
             Self::OvsBridge(ref mut iface) => {
                 iface.pre_verify_cleanup();
             }
+            _ => (),
         }
     }
 
     pub(crate) fn pre_edit_cleanup(&mut self) -> Result<(), NmstateError> {
-        self.base_iface_mut().pre_edit_cleanup()
+        self.base_iface_mut().pre_edit_cleanup()?;
+        if let Interface::MacVlan(iface) = self {
+            iface.validate()?;
+        } else if let Interface::MacVtap(iface) = self {
+            iface.validate()?;
+        }
+        Ok(())
```

Comment:
Why is this required? For the other interfaces we are not validating during the `pre_edit_cleanup`.. I think this should be on `validate()` instead on `src/lib/iface.rs`.

Reply:
Moved to `Interface::validate()`